cli/command/image/build.go

```diff
@@ -344,9 +360,18 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 	// Everything worked so if -q was provided the output from the daemon
 	// should be just the image ID and we'll print that to stdout.
 	if options.quiet {
-		fmt.Fprintf(dockerCli.Out(), "%s", buildBuff)
+		imageID = fmt.Sprintf("%s", buildBuff)
+		fmt.Fprintf(dockerCli.Out(), "%s", imageID)
```

Comment:
> Is passing a potentially untrusted string to a thing which expects a format string always safe in go?

How is it untrusted? You just made it a string on the line above. What I'm suggesting is that there's no reason to format it twice. Either you have:

(formatted as string on line 1)
```go
    imageID = fmt.Sprintf("%s", buildBuff)
    fmt.Fprint(dockerCli.Out(), imageID)
```

or even better

(formatted as string in a single line)
```go
  fmt.Fprintf(dockerCli.Out(), "%s", buildBuff)
```

I quickly looked into the vulnerability you mentioned. From what I can tell it is not an issue in go: https://play.golang.org/p/_Xb8xi2-qP

I would be very surprised if this was an issue in any modern language...